mapfile.c

```diff
@@ -1351,6 +1351,41 @@ int msLoadProjectionStringEPSG(projectionObj *p, const char *value)
   return msLoadProjectionString(p, value);
 }
 
+int msLoadProjectionCodeString(projectionObj* p, const char* value) {
+
+  int num_params = 0;
+
+  // exit if init= is already at the start of the string e.g. from msOGRSpatialRef2ProjectionObj
+  if (strncasecmp(value, "init=", 5) == 0) {
+      return -1;
+  }
+
+  if (!strchr(value, ':')) {
+      return -1;
+  }
+
+  char** papszList = msStringSplit(value, ':', &(num_params));
+
+  if (num_params != 2) {
+      msFreeCharArray(papszList, num_params);
+      return -1;
+  }
+
+  const size_t buffer_size = 5 + strlen(value) + 1;
+  char* init_string = (char*)msSmallMalloc(buffer_size);
+
+  /* translate into PROJ format. */
+  snprintf(init_string, buffer_size, "init=%s:%d", papszList[0], atoi(papszList[1]));
```

Comment:
No reason to restrict that to numeric codes. Some authorities have alphanumeric codes, so:
```suggestion
  snprintf(init_string, buffer_size, "init=%s:%s", papszList[0], papszList[1]);
```